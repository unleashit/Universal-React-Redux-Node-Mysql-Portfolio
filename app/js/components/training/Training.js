import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import StickyHeader from '../stickyHeader/stickyHeader';
import ResponsiveMenu from '../navigation/responsiveMenu';
import TrainingMain from './trainingMain';
import * as globalActions from '../../actions/global';

if (typeof document !== 'undefined')
    require('../../../scss/training/training.scss');

export class Training extends Component {
    static readyOnActions(dispatch, params, bypassCheck = false) {
        return Promise.all([dispatch(globalActions.animateOff())]);
    }

    componentDidMount() {
        const { dispatch, params } = this.props;
        Training.readyOnActions(dispatch, params);
        window.scrollTo(0, 0);
    }

    openBurger() {
        this.props.dispatch(globalActions.openHamburger());
    }

    closeBurger() {
        this.props.dispatch(globalActions.closeHamburger());
    }

    // hide header on small devices when chat is open
    headerVisible() {
        if (typeof window !== 'undefined') {
            return window.innerWidth <= 768
                ? !this.props.liveChat.chatOpen
                : true;
        } else return true;
    }

    render() {
        const { hamburgerState, htmlClass } = this.props.global;
        const htmlClassCheck = htmlClass ? { class: htmlClass } : {};

        const title = 'Front End Development Bootcamp';
        const metadesc =
            'Full Stack Training in Javascript, Html, CSS, React, Node.js, Mysql, Linux and more';

        return (
            <div id="interior-page">
                <Helmet
                    title={title}
                    htmlAttributes={htmlClassCheck}
                    meta={[{ name: 'description', content: metadesc }]}
                    link={[
                        {
                            rel: 'canonical',
                            href: 'https://jasongallagher.org/training',
                        },
                    ]}
                />
                <StickyHeader
                    visible={this.headerVisible()}
                    openBurger={this.openBurger.bind(this)}
                    remoteId={this.props.liveChat.remoteId}
                    dispatch={this.props.dispatch}
                />
                <ResponsiveMenu
                    closeBurger={this.closeBurger.bind(this)}
                    menuVisible={hamburgerState}
                />

                <TrainingMain />
            </div>
        );
    }
}

/* istanbul ignore next */
function mapStateToProps(state) {
    return {
        global: state.global,
        contactForm: state.contactForm,
        liveChat: state.liveChat,
    };
}
/* istanbul ignore next */
function mapDispatchToProps(dispatch) {
    return {
        dispatch: dispatch,
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Training);

Training.propTypes = {
    global: PropTypes.shape({
        hamburgerState: PropTypes.bool.isRequired,
        htmlClass: PropTypes.string,
    }),
    liveChat: PropTypes.shape({
        remoteId: PropTypes.string,
        chatOpen: PropTypes.bool,
    }),
    dispatch: PropTypes.func,
};
