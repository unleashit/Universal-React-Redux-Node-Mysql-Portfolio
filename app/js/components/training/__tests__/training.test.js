import { Training } from '../Training';
import StickyHeader from '../../stickyHeader/stickyHeader';

window.scrollTo = jest.fn();

describe('<Training />', () => {
    let wrapper;
    let props = {
        global: {
            hamburgerState: false,
            htmlClass: '',
        },
        liveChat: {
            remoteId: 'admin',
            chatOpen: false,
        },
        dispatch: jest.fn(),
        params: {},
    };

    beforeEach(() => {
        props.dispatch.mockReset();
        wrapper = shallow(<Training {...props} />);
    });

    describe('Sections', () => {
        it('renders <Helmet />', () => {
            expect(wrapper.find('HelmetWrapper')).toHaveLength(1);
        });
        it('renders <StickyHeader />', () => {
            expect(wrapper.find(StickyHeader)).toHaveLength(1);
        });
        it('renders <ResponsiveMenu />', () => {
            expect(wrapper.find('ResponsiveMenu')).toHaveLength(1);
        });
        it('renders <TrainingMain />', () => {
            expect(wrapper.find('TrainingMain')).toHaveLength(1);
        });
    });

    test('snapshot matches', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('openBurger()', () => {
        props.dispatch.mockReset();
        wrapper.instance().openBurger();
        expect(props.dispatch).toHaveBeenCalledTimes(1);
    });

    test('closeBurger()', () => {
        props.dispatch.mockReset();
        wrapper.instance().closeBurger();
        expect(props.dispatch).toHaveBeenCalledTimes(1);
    });

    test('passes html attributes to helmet', () => {
        props.global.htmlClass = 'test';
        wrapper = shallow(<Training {...props} />);

        const helmet = wrapper.find('HelmetWrapper');
        expect(helmet.prop('htmlAttributes')).toEqual({
            class: props.global.htmlClass,
        });
    });

    test('headerVisible()', () => {
        let result;
        props.liveChat.chatOpen = true;
        wrapper = shallow(<Training {...props} />);

        global.innerWidth = 500;
        result = wrapper.instance().headerVisible();
        expect(result).toBe(false);

        global.innerWidth = 800;
        result = wrapper.instance().headerVisible();
        expect(result).toBe(true);

        // simulate code running on server/node.js
        // this must come last
        delete global.window;
        result = wrapper.instance().headerVisible();
        expect(result).toBe(true);
    });
});
