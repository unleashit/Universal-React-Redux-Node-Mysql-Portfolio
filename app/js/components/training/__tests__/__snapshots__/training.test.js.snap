// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Training /> snapshot matches 1`] = `
<div
  id="interior-page"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    htmlAttributes={Object {}}
    link={
      Array [
        Object {
          "href": "https://jasongallagher.org/training",
          "rel": "canonical",
        },
      ]
    }
    meta={
      Array [
        Object {
          "content": "Full Stack Training in Javascript, Html, CSS, React, Node.js, Mysql, Linux and more",
          "name": "description",
        },
      ]
    }
    title="Front End Development Bootcamp"
  />
  <Connect(StickyHeader)
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    openBurger={[Function]}
    remoteId="admin"
    visible={true}
  />
  <ResponsiveMenu
    closeBurger={[Function]}
    menuVisible={false}
  />
  <TrainingMain />
</div>
`;
